squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/impl/FrontierImpl.java

```diff
@@ -347,10 +328,8 @@ public void crawlingDone(List<CrawleableUri> uris) {
 
     @Override
     public int getNumberOfPendingUris() {
-        // TODO this implementation does not fit to the semantics of the method name
-        // since it returns the number of URI groups instead of the number of URIs
-        if (queue instanceof BlockingQueue) {
-            return ((BlockingQueue<?>) queue).getNumberOfBlockedKeys();
+        if (queue instanceof IpAddressBasedQueue) {
+            return ((IpAddressBasedQueue) queue).getNumberOfBlockedIps();
```

Comment:
This seems to re-introduce old code that we removed in previous commits. Please double check.